src/query/src/optimizer/scan_hint.rs

```diff
@@ -121,6 +121,30 @@ impl ScanHintRule {
             });
         }
         adapter.with_ordering_hint(&opts);
+
+        let mut sort_expr_cursor = order_expr.iter().map(|s| s.expr.try_as_col()).flatten();
+        let region_metadata = adapter.region_metadata();
+        let mut pk_column_iter = region_metadata.primary_key_columns();
+        let mut curr_sort_expr = sort_expr_cursor.next();
+        let mut curr_pk_col = pk_column_iter.next();
+
+        while let (Some(sort_expr), Some(pk_col)) = (curr_sort_expr, curr_pk_col) {
+            if sort_expr.name == pk_col.column_schema.name {
+                curr_sort_expr = sort_expr_cursor.next();
+            }
+            curr_pk_col = pk_column_iter.next();
```

Comment:
This hint may not be suitable for non-metric situations.
If a table has pk `a, b, c, ...` and the following rows:
```
a, b, c
1, 1, 1
1, 1, 2
1, 2, 1
1, 2, 2
```
When the sort expr is `a, c, ts`, maybe time series distribution doesn't have benefits.

Reply:
I change the check to require order expr to be exact the same with pk